purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -0,0 +1,158 @@
+
+        Make this compatible with other extensions, only subtracting refunds
+        from the number obtained in super.
+        """
+        super(PurchaseOrder, self)._compute_invoice()
+        for order in self:
+            order.invoice_count -= order.invoice_refund_count
+
+    @api.multi
+    def action_view_invoice_refund(self):
+        """This function returns an action that display existing vendor refund
+        bills of given purchase order id.
+        When only one found, show the vendor bill immediately.
+        """
+        action = self.env.ref('account.action_invoice_tree2')
+        result = action.read()[0]
+        refunds = self.invoice_ids.filtered(lambda x: x.type == 'in_refund')
+        # override the context to get rid of the default filtering
+        result['context'] = {'type': 'in_refund',
+                             'default_purchase_id': self.id}
+        if not refunds:
+            # Choose a default account journal in the
+            # same currency in case a new invoice is created
+            journal_domain = [
+                ('type', '=', 'purchase'),
+                ('company_id', '=', self.company_id.id),
+                ('currency_id', '=', self.currency_id.id),
+            ]
+            default_journal_id = self.env['account.journal'].search(
+                journal_domain, limit=1)
+            if default_journal_id:
+                result['context']['default_journal_id'] = default_journal_id.id
+        else:
+            # Use the same account journal than a previous invoice
+            result['context']['default_journal_id'] = refunds[0].journal_id.id
+        # choose the view_mode accordingly
+        if len(refunds) != 1:
+            result['domain'] = [('id', 'in', refunds.ids)]
+        elif len(refunds) == 1:
+            res = self.env.ref('account.invoice_supplier_form', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            result['res_id'] = refunds.id
+        return result
+
+    @api.multi
+    def action_view_invoice(self):
+        """Change super action for displaying only normal invoices."""
+        result = super(PurchaseOrder, self).action_view_invoice()
+        invoices = self.invoice_ids.filtered(
+            lambda x: x.type == 'in_invoice'
+        )
+        # choose the view_mode accordingly
+        if len(invoices) != 1:
+            result['domain'] = [('id', 'in', invoices.ids)]
+        elif len(invoices) == 1:
+            res = self.env.ref('account.invoice_supplier_form', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            result['res_id'] = invoices.id
+        return result
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    qty_refunded = fields.Float(
+        compute="_compute_qty_refunded",
+        string='Refunded Qty',
+    )
+    qty_returned = fields.Float(
+        compute="_compute_qty_returned",
+        string='Returned* Qty',
```

Comment:
Why the asterisk?

Reply:
It's explained in the help. It's not exactly the returned qty, but the refundable returned qty